modules/fastani/main.nf

```diff
@@ -0,0 +1,46 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process FASTANI {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::fastani=1.32" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/fastani:1.32--he1c1bb9_0"
+    } else {
+        container "quay.io/biocontainers/fastani:1.32--he1c1bb9_0"
+    }
+
+    input:
+    val(meta)
```

Comment:
Why is `val meta` a stand-alone variable here? The convention is to associate sample info like this with a file of some sort e.g. `tuple val(meta), path(query)` so `meta` can be carried through the pipeline withot messing around with channels too much.

Reply:
Ah, yes. This is a good point.

I have accomodate the suggestion w.r.t the `query` variable

Reply:
Thanks for the review @drpatelh 😊